### PR TITLE
Added framework benchmark collecting code size and compile time

### PR DESCRIPTION
**Pull request by LepilkinaElena**

### CODE REVIEW

File: performance/framework/build.gradle [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            kotlin.srcDir "$toolsPath/kliopt"
+            kotlin.srcDir "$toolsPath/benchmarksAnalyzer/src/main/kotlin-native"
+        }
+    }
+
+    configure([macosX64("macos")]) {
+        compilations.main.cinterops {
+            libcurl {
+                defFile "$toolsPath/benchmarksAnalyzer/src/nativeInterop/cinterop/libcurl.def"
+                includeDirs.headerFilterOnly '/opt/local/include', '/usr/local/include'
```
Comment (SvyatoslavScherbina) [inline performance/framework/build.gradle]: Not sure if this is required for libcurl on macOS.

File: performance/buildSrc/src/main/kotlin/MPPTools.kt [hunk 2]
```diff
@@ -84,8 +84,21 @@ fun getFileSize(filePath: String): Long? {
     return if (file.exists()) file.length() else null
 }
 
-fun getCodeSizeBenchmark(programName: String, filePath: String): BenchmarkResult {
-    val codeSize = getFileSize(filePath)
+fun getFolderSize(folderPath: String): Long? {
```
Comment (SvyatoslavScherbina) [inline performance/buildSrc/src/main/kotlin/MPPTools.kt]: Btw, frameworks on macOS have symlinks inside. Does this function support symlinks properly?

File: performance/buildSrc/src/main/kotlin/MPPTools.kt [hunk 3]
```diff
@@ -84,8 +84,21 @@ fun getFileSize(filePath: String): Long? {
...
+    var length = 0L
+    folder.listFiles().forEach { file ->
+        length += if (file.isFile()) file.length() else getFolderSize(file.absolutePath) ?: 0
+    }
+    return length
+}
+
+@JvmOverloads
+fun getCodeSizeBenchmark(programName: String, filePath: String, isFramework: Boolean = false): BenchmarkResult {
+    val codeSize = if (isFramework) getFolderSize(filePath) else getFileSize(filePath)
```
Comment (SvyatoslavScherbina) [inline performance/buildSrc/src/main/kotlin/MPPTools.kt]: I can't see why should the report include framework header size too.

File: performance/framework/build.gradle [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+}
+
+task jvmRun(type: RunJvmTask)  {
+    doLast {
+        println("JVM run is unsupported")
+    }
+}
+
+task konanJsonReport {
```
Comment (olonho) [inline performance/framework/build.gradle]: Shalln't we use .kts for the new non-trivial code?

File: tools/performance-server/ui/index.ejs [hunk 5]
```diff
@@ -11,7 +11,7 @@
     </script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js">
     </script>
-    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js">
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js">
```
Comment (olonho) [inline tools/performance-server/ui/index.ejs]: Maybe cache those locally?

File: tools/performance-server/ui/src/main/kotlin/main.kt [hunk 6]
```diff
@@ -228,7 +228,7 @@ fun main(args: Array<String>) {
     val branchesUrl = "$serverUrl/branches/${parameters["target"]}"
 
     val branches: Array<String> = JSON.parse(sendGetRequest(branchesUrl))
-    val releaseBranches = branches.filter { "v\\d+\\.\\d+\\.\\d+-fixes".toRegex().find(it) != null }
+    val releaseBranches = branches.filter { "^v\\d+\\.\\d+\\.\\d+-fixes$".toRegex().find(it) != null }
```
Comment (olonho) [inline tools/performance-server/ui/src/main/kotlin/main.kt]: Thanks

File: performance/framework/build.gradle (hunk 4 above)
Comment (LepilkinaElena) [reply performance/framework/build.gradle]: It isn't fully separated file, it's part of performance project. To use Kotlin in Gradle build files, I have to rewrite full project. I have already done such attempt, but there were problems with current project.

File: tools/performance-server/ui/index.ejs (hunk 5 above)
Comment (LepilkinaElena) [reply tools/performance-server/ui/index.ejs]: What do you mean? All js and css files are cached by browser by default.

**Pull request merged**